### DISCUSSION TITLE
Mat/Affine dimensional truncation

### DISCUSSION THREAD

**Discussion by Demiu** [bitshifter/glam-rs#323] [Ideas]
Hello, while working with 2D/3D transformation using matrices I felt a certain transformation was missing. There is no ergonomic way to turn a matrix for 3D transformation (Mat4) into a matrix for 2D transformation (Mat3) preserving the translation; for example cut off Z off a Mat4, so remove 3rd column and 3rd row. Same thing is missing from the Affine types.

I think it would be a good convenience function

**Comment by bitshifter** [comment]
What would that mean for rotation?

**Reply by bitshifter** [reply]
There is a matrix function called minor that I implemented in a different project that would probably do the job, see https://github.com/bitshifter/bevy-physics-weekend/blob/main/physics/src/math/glam_ext.rs#L9-L30.


https://en.m.wikipedia.org/wiki/Minor_(linear_algebra)

**Reply by Demiu** [reply]
That's it, minor is the more general name for the thing I was thinking

**Comment by bitshifter** [comment]
Created an issue for this #325
